tests/mocha/workspace_svg_test.js

```diff
@@ -116,6 +116,171 @@ suite('WorkspaceSvg', function() {
     });
   });
 
+  suite.only('Viewport change events', function() {
+    function resetEventHistory(eventsFireStub, changeListenerSpy) {
+      eventsFireStub.resetHistory();
+      changeListenerSpy.resetHistory();
+    }
+    function assertSpyFiredViewportEvent(spy, workspace, expectedProperties) {
+      assertEventFired(
+          spy, Blockly.Events.Ui, {element: 'viewport'},
+          workspace.id, null);
+      assertEventFired(spy, Blockly.Events.Ui, expectedProperties,
+          workspace.id, null);
+    }
+    function assertViewportEventFired(eventsFireStub, changeListenerSpy,
+        workspace, expectedEventCount = 1) {
+      var metrics = workspace.getMetrics();
+      var expectedProperties = {
+        element: 'viewport',
+        newValue: {scale: workspace.scale, top: metrics.viewTop,
+          left: metrics.viewLeft}
+      };
+      assertSpyFiredViewportEvent(
+          eventsFireStub, workspace, expectedProperties);
+      assertSpyFiredViewportEvent(
+          changeListenerSpy, workspace,expectedProperties);
+      sinon.assert.callCount(changeListenerSpy, expectedEventCount);
+      sinon.assert.callCount(eventsFireStub, expectedEventCount);
+    }
+    function runViewportEventTest(eventTriggerFunc, eventsFireStub,
+        changeListenerSpy, workspace, clock, expectedEventCount = 1) {
+      clock.runAll();
+      resetEventHistory(eventsFireStub, changeListenerSpy, this.clock);
+      eventTriggerFunc();
+      assertViewportEventFired(
+          eventsFireStub, changeListenerSpy, workspace, expectedEventCount);
+    }
+    setup(function() {
+      defineStackBlock();
+      this.changeListenerSpy = createFireChangeListenerSpy(this.workspace);
+    });
+    teardown(function() {
+      delete Blockly.Blocks['stack_block'];
+    });
+
+    test('inject', function() {
```

Comment:
Should there be a test for event behavior with inject call? Things are in a weird state where events are fired, but change listeners are not attached during inject call.